webviz_subsurface/plugins/_group_tree/_plugin.py

```diff
@@ -66,39 +72,47 @@ def __init__(
         ], "time_index must be monthly or yearly"
         self.ensembles = ensembles
         self.gruptree_file = gruptree_file
-        self.time_index = time_index
-        self.ens_paths = {
-            ens: webviz_settings.shared_settings["scratch_ensembles"][ens]
-            for ens in ensembles
+
+        if ensembles is None:
+            raise ValueError('Incorrect argument, must provide "ensembles"')
+
+        self._sampling = Frequency(time_index)
+
+        self.ensemble_paths: Dict[str, Path] = {
+            ensemble_name: webviz_settings.shared_settings["scratch_ensembles"][
+                ensemble_name
+            ]
+            for ensemble_name in ensembles
         }
 
-        self.emodel: EnsembleSetModel = (
-            caching_ensemble_set_model_factory.get_or_create_model(
-                ensemble_paths={
-                    ens: webviz_settings.shared_settings["scratch_ensembles"][ens]
-                    for ens in ensembles
-                },
-                time_index="monthly",
-            )
-        )
-        smry = self.emodel.get_or_load_smry_cached()
-        gruptree = read_gruptree_files(self.ens_paths, self.gruptree_file)
-        smry["DATE"] = pd.to_datetime(smry["DATE"])
-        gruptree["DATE"] = pd.to_datetime(gruptree["DATE"])
+        provider_factory = EnsembleSummaryProviderFactory.instance()
 
-        if time_index == "yearly":
-            smry = smry[smry["DATE"].dt.is_year_start]
+        self.ensemble_dict: Dict[str, EnsembleGroupTreeData] = {}
 
-        self.grouptreedata = GroupTreeData(smry, gruptree)
+        for ens_name, ens_path in self.ensemble_paths.items():
+            provider: EnsembleSummaryProvider = (
+                provider_factory.create_from_arrow_unsmry_presampled(
+                    str(ens_path), rel_file_pattern, self._sampling
+                )
+            )
+            gruptree = read_ensemble_gruptree(ens_name, ens_path, gruptree_file)
+            self.ensemble_dict[ens_name] = EnsembleGroupTreeData(provider, gruptree)
```

Comment:
Consider protected attribute notation for protected GroupTree-class attributes, i.e. underscore prefix: `self._ensembles`, `self._ensemble_dict`, etc.